### PR TITLE
Add is_deprecated and get_non_deprecated() to CRS

### LINKED ISSUE #1367: Support for deprecated and non-deprecated objects
Initially from https://github.com/OSGeo/spatialreference.org/issues/3#issuecomment-1881859037 by @rouault it would be helpful to have support to see deprecated objects, and provide lists to non-deprecated objects.

For instance, here is a prototype of possible behavior to add a `.is_deprecated` property and a `get_non_deprecated` function:

```python
import pyproj
from pyproj import CRS

crs1 = CRS.from_epsg(28473)
assert crs1.is_deprecated
crs1_nondep = pyproj.get_non_deprecated(crs1)
assert len(crs1_nondep) == 1
assert crs1_nondep[0] == CRS.from_epsg(2503)

crs2 = CRS.from_epsg(4326)
assert not crs2.is_deprecated
assert len(pyproj.get_non_deprecated(crs2)) == 0
```

Relevant PROJ C API:

- [`proj_get_non_deprecated`](https://proj.org/en/9.3/development/reference/functions.html#c.proj_get_non_deprecated)
- [`proj_is_deprecated`](https://proj.org/en/9.3/development/reference/functions.html#c.proj_is_deprecated)

### PR DESCRIPTION
**Pull request by jjimenezshaw**
This PR adds two methods:

- is_deprecated
- get_non_deprecated

To replicate the functionality in PROJ C API.

- [`proj_is_deprecated`](https://proj.org/en/9.3/development/reference/functions.html#c.proj_is_deprecated)
- [`proj_get_non_deprecated`](https://proj.org/en/9.3/development/reference/functions.html#c.proj_get_non_deprecated)

 - [x] Closes #1367
 - [x] Tests added
 - [x] Fully documented, including `history.rst` for all changes and `api/*.rst` for new API

### CODE REVIEW

File: pyproj/_crs.pyx [hunk 1]
```diff
@@ -3203,3 +3203,60 @@ cdef class _CRS(Base):
         if not isinstance(other, _CRS):
             return False
         return self._equals(other, ignore_axis_order=ignore_axis_order)
+
+    @property
+    def is_deprecated(self):
+        """
+        .. versionadded:: 3.6.2
```
Comment (snowman2) [inline pyproj/_crs.pyx:3210]:
```suggestion
        .. versionadded:: 3.7.0
```

File: pyproj/_crs.pyx [hunk 2]
```diff
@@ -3203,3 +3203,60 @@ cdef class _CRS(Base):
...
+
+        Returns
+        -------
+        bool
+        """
+        return bool(proj_is_deprecated(self.projobj))
+
+    def get_non_deprecated(self):
+        """
+        .. versionadded:: 3.6.2
```
Comment (snowman2) [inline pyproj/_crs.pyx:3222]:
```suggestion
        .. versionadded:: 3.7.0
```

**Comment by snowman2** [comment]
Looks great 👍. Seems like there are unrelated failures. So, should be good to go after the minor changes.

File: pyproj/crs/crs.py [hunk 3]
```diff
@@ -1565,6 +1565,31 @@ def is_derived(self):
         """
         return self._crs.is_derived
 
+    @property
+    def is_deprecated(self) -> bool:
+        """
+        .. versionadded:: 3.6.2
```
Comment (jjimenezshaw) [inline pyproj/crs/crs.py:1571]:
```suggestion
        .. versionadded:: 3.7.0
```

File: pyproj/crs/crs.py [hunk 4]
```diff
@@ -1565,6 +1565,31 @@ def is_derived(self):
...
+
+        Returns
+        -------
+        bool
+        """
+        return self._crs.is_deprecated
+
+    def get_non_deprecated(self) -> list["CRS"]:
+        """
+        .. versionadded:: 3.6.2
```
Comment (jjimenezshaw) [inline pyproj/crs/crs.py:1583]:
```suggestion
        .. versionadded:: 3.7.0
```

**Pull request merged**